docs/design.md

```diff
@@ -8,96 +8,194 @@ Requirements for individual releases can be found on the [Project Page](https://
 
+XXX: Docker v1.6 is very old. Perhaps this is no longer useful and can be removed now?
+
 ## Goal
 
-libnetwork project will follow Docker and Linux philosophy of developing small, highly modular and composable tools that work well independently.
-Libnetwork aims to satisfy that composable need for Networking in Containers.
+The libnetwork project follows the Docker and Linux philosophy of developing small, highly modular and composable tools that work well independently.
+libnetwork aims to satisfy the need for composable networking in containers.
 
 ## The Container Network Model
 
-Libnetwork implements Container Network Model (CNM) which formalizes the steps required to provide networking for containers while providing an abstraction that can be used to support multiple network drivers. The CNM is built on 3 main components (shown below)
+libnetwork implements the Container Network Model (CNM) which formalizes the steps required to provide networking for containers while providing an abstraction that can be used to support multiple network drivers. The CNM is built on 3 main components (shown below)
 
 ![](/docs/cnm-model.jpg?raw=true)
 
+XXX: This diagram is really confusing when considering overlay networks (which are the default).
+There, as I understand it, you have both a "container sandbox" (which the diagram calls the "Network Sandbox") and a separate network sandbox (which connects the network to all of that network's containers on a single host).
+
 **Sandbox**
 
 A Sandbox contains the configuration of a container's network stack.
 This includes management of the container's interfaces, routing table and DNS settings.
 An implementation of a Sandbox could be a Linux Network Namespace, a FreeBSD Jail or other similar concept.
 A Sandbox may contain *many* endpoints from *multiple* networks.
+The sandbox is managed by libnetwork itself, not by any individual driver.
 
 **Endpoint**
 
 An Endpoint joins a Sandbox to a Network.
 An implementation of an Endpoint could be a `veth` pair, an Open vSwitch internal port or similar.
-An Endpoint can belong to *only one* network but may only belong to *one* Sandbox.
+An Endpoint can belong to *only one* network and to *only one* Sandbox.
 
 **Network**
 
 A Network is a group of Endpoints that are able to communicate with each-other directly.
 An implementation of a Network could be a Linux bridge, a VLAN, etc.
 Networks consist of *many* endpoints.
 
-## CNM Objects
+## Services
+
+A Service (e.g. a SwarmKit service) is an abstract concept used for load balancing and discovery.
+Each service has a number of replicas, each with its own Endpoint.
+Each endpoint has its own IP address, and the service itself has another IP address, called its Virtual IP (VIP).
+When an application connects to the service's VIP, libnetwork will load-balance this across the available endpoints.
+
+## Go API
```

Comment:
This section was titled `CNM Objects`, but that didn't mean anything to me. And e.g. `Network` here referred back to `CNM : Network`, which was pretty confusing.